data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -78,41 +90,58 @@ public WebClientCloudEventSender(final WebClient client,
     this.client = client;
     this.target = target;
     this.circuitBreaker = circuitBreaker;
+    this.egress = egress;
+    this.retryPolicyFunc = ConsumerVerticleFactoryImpl.computeRetryPolicy(egress);
   }
 
-  @Override
-  public Future<HttpResponse<Buffer>> send(CloudEvent event) {
+  public Future<HttpResponse<Buffer>> send(final CloudEvent event) {
+    return send(event, 0);
+  }
+
+  private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retryCounter) {
     logger.debug("Sending event {} {}",
       keyValue("id", event.getId()),
       keyValue("subscriberURI", target)
     );
 
-    TracingSpan.decorateCurrentWithEvent(event);
 
-    final Future<HttpResponse<Buffer>> future = circuitBreaker.execute(breaker -> {
-      if (closed.get()) {
-        // Once sender is closed, return a successful future to avoid retrying.
-        breaker.tryComplete(null);
-        return;
-      }
+    final Promise<HttpResponse<Buffer>> promise = Promise.promise();
 
+    if (closed.get()) {
+      // Once sender is closed, return a successful future to avoid retrying.
+      promise.tryComplete(null);
+    } else {
       try {
+        TracingSpan.decorateCurrentWithEvent(event);
         requestEmitted();
-        send(event, breaker).
+        send(event, promise).
           onComplete(v -> requestCompleted());
       } catch (CloudEventRWException e) {
         logger.error("failed to write event to the request {}", keyValue("subscriberURI", target), e);
-        breaker.tryFail(e);
+        promise.tryFail(e);
       }
-    });
+    }
 
     // Handle closed return value.
-    return future.compose(r -> {
-      if (r == null) {
-       return Future.failedFuture("Sender closed for target="+target);
+    return promise.future().compose(
+      r -> {
+        if (r == null) {
+          return Future.failedFuture("Sender closed for target=" + target);
+        }
+        return Future.succeededFuture(r);
+      },
+      // Retry and failures
+      cause -> {
+        if (retryCounter+1 > egress.getRetry()) {
+          return Future.failedFuture(cause);
+        }
+
+        Promise<HttpResponse<Buffer>> r = Promise.promise();
+        final var delay = retryPolicyFunc.apply(retryCounter + 1);
+        vertx.setTimer(delay, v -> send(event, retryCounter + 1).onComplete(r));
```

Comment:
Do we need to take the timeout into account here ?

Reply:
Timeout is set on the request itself, see line 161

Reply:
yeah, via the breaker options, that;s why I added the other comment :-)

Reply:
ah ok, so yes, we're still configuring the CB options